toolchain/language_server/context.cpp

```diff
@@ -155,6 +155,55 @@ auto Context::File::SetText(Context& context, std::optional<int64_t> version,
                             consumer.params());
 }
 
+auto Context::File::PositionToIndex(const clang::clangd::Position& position)
+    -> size_t {
+  return PositionToIndex(source_->text().data(), position);
+}
+
+auto Context::File::PositionToIndex(const std::string& contents,
+                                    const clang::clangd::Position& position)
+    -> size_t {
+  size_t result = 0;
+
+  for (int row = 0; row < position.line; row++) {
```

Comment:
I think we can use `llvm::seq` where to make clear this is just doing N finds:
```suggestion
  for ([[maybe_unused]] auto _ : llvm::seq(position.line)) {
```